data/org.eclipse.birt.report.data.oda.jdbc.ui/src/org/eclipse/birt/report/data/oda/jdbc/ui/provider/JdbcMetaDataProvider.java

```diff
@@ -707,13 +709,43 @@ public ResultSet getAllSchemas( )
 			public void run( )
 			{
 				ResultSet rs = JdbcMetaDataProvider.this.getAllSchemas( );
+				//As it�s Cassandra�s behavior to display all tables in all namespaces,
+				//user can pass "OTSCPF" as connection params inorder to  restrict to view in UI.
+				String odaDriver = JdbcMetaDataProvider.this.driverClass.toLowerCase();
+				boolean isCassandra = ( odaDriver !=null && odaDriver.indexOf("cassandra") != -1 );
```

Comment:
Abstract JDBC layer should not check hardcoded name or depend on specific JDBC driver.
This logic should push down to the extension layer like jdbc.cassandra.
More code refactor is required.

Reply:
This logic cannot be moved to Data Direct cassandra driver. 
it  is better not to fix it for Designer.

Reply:
You need to code refactor the jdbc UI (super class) to allow Cassandra layer (subclass) to override, so that specific logic only exists in specific plugin.